### PR TITLE
Create component for Ingredient Card

### PR DESCRIPTION
**Pull request by vreddi**

### CODE REVIEW

File: src/assets/json/themes.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  "teal": "teal",
+  "hotPink": "hotPink",
+  "lime": "lime",
+  "default": ""
+}
```
Comment (sukritchhabra) [inline src/assets/json/themes.json]: incorrect line ending

File: src/components/Chip/Chip.scss [hunk 2]
```diff
@@ -0,0 +1,11 @@
+.chip {
+    display: inline-flex;
+    justify-content: center;
+    align-items: center;
+    padding: 0 0.4em;
+    height: 1.2em;
+    font-size: 0.8em;
+    line-height: 3em;
+    border-radius: 1.6em;
+    background-color: #f1f1f1;
```
Comment (sukritchhabra) [inline src/components/Chip/Chip.scss]: Make this color a variable. Either in this file or in _colors

File: src/components/Dashboard/Dashboard.jsx [hunk 3]
```diff
@@ -1,11 +1,36 @@
...
+import Chip from 'components/Chip/Chip';
 
+import Milk from 'images/ingredients/milk.png';
+import Yogurt from 'images/ingredients/yogurt.png';
+
+// TODO: Move all the place components in their own area
 const Dashboard = () => (
   <div className="dashboard">
     <Header />
+    <IngredientCard
```
Comment (sukritchhabra) [inline src/components/Dashboard/Dashboard.jsx:16]: What is your plane on deciding these themes in the future?? This is why i was suggesting that you should be getting the theme from services. How are you going to decide the them when you get the Name and Image from services?

File: src/components/Dashboard/Dashboard.jsx [hunk 4]
```diff
@@ -1,11 +1,36 @@
...
+      name="Milk"
+      image={Milk}
+      theme={Themes.teal}
+    >
+      <Chip name="calcium" />
```
Comment (sukritchhabra) [inline src/components/Dashboard/Dashboard.jsx:21]: You should create template json files for this! You should run a map over the array in the json and get these so you dont have to write each one separately

File: src/components/IngredientCard/IngredientCard.jsx [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+import './IngredientCard.scss';
+
+const IngredientCard = ({
+  name,
+  image,
+  theme,
+  children,
+}) => (
+  <div className={`ingredient-card ${Themes[theme] || Themes.default}`}>
```
Comment (sukritchhabra) [inline src/components/IngredientCard/IngredientCard.jsx:13]: Use of `Theme[theme]` here is unnecessary! You're already passing `Theme.teal etc` to this comp

File: src/components/Dashboard/Dashboard.jsx (hunk 3 above)
Comment (vreddi) [reply src/components/Dashboard/Dashboard.jsx:16]: The theme will come from the backend. I don't know if you remember but we talked about this earlier.

File: src/components/Dashboard/Dashboard.jsx (hunk 4 above)
Comment (vreddi) [reply src/components/Dashboard/Dashboard.jsx:21]: These are properties on the Ingredient model and will come from the backend. This is just a prototype display. The dashboard component will not be like that. In the development stage for components it is just a display.

File: src/components/IngredientCard/IngredientCard.jsx (hunk 5 above)
Comment (vreddi) [reply src/components/IngredientCard/IngredientCard.jsx:13]: This needs to be there. The backend will provide the theme to the frontend. And that will be the key. Currently I am using Themes.teal on `Dashboard` till the backend is done. `IngredientCard` would still have this code.

Comment (sukritchhabra) [reply src/components/IngredientCard/IngredientCard.jsx:13]: My point being, over here you will **`always`** get a value cause you're passing it from the parent. Use of default should be moved to the parent.

Comment (vreddi) [reply src/components/IngredientCard/IngredientCard.jsx:13]: @sukritchhabra But someone creating a component might not pass that prop. It is not a required prop so we need a default color on the component right?

Comment (sukritchhabra) [reply src/components/IngredientCard/IngredientCard.jsx:13]: We should mark the `theme` prop required using PropTypes in that case too

File: src/components/IngredientCard/IngredientCard.scss [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+            width: auto;
+            height: 100%;
+            overflow: hidden;
+            text-align: center;
+            vertical-align: middle;
+        }
+    }
+
+    .ingredient-body {
+        display: flex;
```
Comment (sukritchhabra) [inline src/components/IngredientCard/IngredientCard.scss]: Again why do we need flex? we can do this with text-align

File: src/components/IngredientCard/IngredientCard.scss [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+    flex-direction: column;
+    width: 9em;
+    height: 8em;
+    border-radius: 0.6em;
+    overflow: hidden;
+    padding: 0.6em;
+    box-shadow: 0 0.4em 0.8em 0 rgba(0,0,0,0.2);
+
+    .ingredient-header {
+        display: flex;
```
Comment (sukritchhabra) [inline src/components/IngredientCard/IngredientCard.scss]: Don't need `flex` here

File: src/components/IngredientCard/IngredientCard.scss [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+        height: 40%;
+
+        .ingredient-image {
+            display: inline-block;
+            width: auto;
+            height: 100%;
+        }
+    
+        .ingredient-name {
+            display: flex;
```
Comment (sukritchhabra) [inline src/components/IngredientCard/IngredientCard.scss]: This literally just holds text. You should almost never need flex here.

File: src/components/IngredientCard/IngredientCard.scss [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+    box-shadow: 0 0.4em 0.8em 0 rgba(0,0,0,0.2);
+
+    .ingredient-header {
+        display: flex;
+        justify-content: start;
+        width: 100%;
+        height: 40%;
+
+        .ingredient-image {
+            display: inline-block;
```
Comment (sukritchhabra) [inline src/components/IngredientCard/IngredientCard.scss]: You can make this a span to achieve this

File: src/components/IngredientCard/IngredientCard.jsx [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+    <div className="ingredient-header">
+      <img
+        className="ingredient-image"
+        alt={name}
+        src={image}
+      />
+      <div className="ingredient-name">
+        {name}
```
Comment (sukritchhabra) [inline src/components/IngredientCard/IngredientCard.jsx]: Why the new line here? We can add it on the same line

File: src/components/IngredientCard/IngredientCard.scss [hunk 11]
```diff
@@ -0,0 +1,53 @@
+.ingredient-card {
```
Comment (sukritchhabra) [inline src/components/IngredientCard/IngredientCard.scss:1]: Please use 2 spaces throughout the project! You're using 4 spaces here

File: src/components/Chip/Chip.scss [hunk 12]
```diff
@@ -0,0 +1,13 @@
+@import '~assets/scss/colors';
+
```
Comment (sukritchhabra) [inline src/components/Chip/Chip.scss:2]: Use 2 spaces

File: src/base.scss [hunk 13]
```diff
@@ -19,4 +19,16 @@ html {
             }
         }
     }
-}
\ No newline at end of file
+}
+
```
Comment (sukritchhabra) [inline src/base.scss]: Use 2 spaces!

File: src/components/Chip/Chip.jsx [hunk 14]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
```
Comment (sukritchhabra) [inline src/components/Chip/Chip.jsx:1]: Why do we call this Chip? Seems like we should come up with a better name

File: src/components/IngredientCard/IngredientCard.scss [hunk 15]
```diff
@@ -0,0 +1,53 @@
...
+    display: flex;
```
Comment (sukritchhabra) [inline src/components/IngredientCard/IngredientCard.scss]: We're using a lot of unnecessary `flex`.

You can do everything you've done while avoid a few flexes.

```scss
.ingredient-card {
    // display: flex;
    // flex-direction: column;
    width: 9em;
    height: 8em;
    border-radius: 0.6em;
    overflow: hidden;
    padding: 0.6em;
    box-shadow: 0 0.4em 0.8em 0 rgba(0,0,0,0.2);

    .ingredient-header {
        display: flex;
        justify-content: start;
        align-items: center;
        width: 100%;
        height: 40%;

        .ingredient-image {
            // display: inline-block;
            width: auto;
            height: 100%;
        }

        .ingredient-name {
            // display: flex;
            // flex-direction: column;
            // justify-content: center;
            color: white;
            font-size: 18px;
            text-transform: uppercase;
            // width: auto;
            // height: 100%;
            overflow: hidden;
            text-align: center;
            vertical-align: middle;
        }
    }

    .ingredient-body {
        display: flex;
        flex-direction: column;
        justify-content: flex-end;
        width: 100%;
        height: 60%;

        .ingredient-tags {
            display: inline-block;

            > * {
                margin: 0px 0.15em;
            }
        }
    }
}
```

Note: Will also need to change `ingredient-name` to a span for the above css. Which makes sense for inline ements. Should use span for inline elements.

File: src/components/Chip/Chip.scss [hunk 16]
```diff
@@ -0,0 +1,13 @@
...
+.chip {
+    display: inline-flex;
```
Comment (sukritchhabra) [inline src/components/Chip/Chip.scss]: Avoid unnecessary flex

chip flex

File: src/components/Chip/Chip.jsx [hunk 17]
```diff
@@ -0,0 +1,19 @@
...
+import PropTypes from 'prop-types';
+import './Chip.scss';
+
+const Chip = ({ name }) => (
+  <div className="chip">
```
Comment (sukritchhabra) [inline src/components/Chip/Chip.jsx]: use span

Comment (sukritchhabra) [reply src/components/Chip/Chip.jsx]: also use one line

File: src/base.scss [hunk 18]
```diff
@@ -19,4 +19,16 @@ html {
...
+.teal {
```
Comment (sukritchhabra) [inline src/base.scss]: Are these final? They dont look great

**Review by sukritchhabra — CHANGES_REQUESTED** [review]

File: src/base.scss (hunk 18 above)
Comment (vreddi) [reply src/base.scss]: No none of the colors/themes are final.

File: src/components/Chip/Chip.jsx (hunk 14 above)
Comment (vreddi) [reply src/components/Chip/Chip.jsx:1]: Chip is the exact term for that component. Tags is another close one but many style libraries including materialize calls it Chip.

File: src/components/IngredientCard/IngredientCard.jsx (hunk 5 above)
Comment (vreddi) [reply src/components/IngredientCard/IngredientCard.jsx:13]: That is a weak argument to make something required. `theme` should not be required to render the component. By providing default in the component the intension is to make it optional.

Comment (sukritchhabra) [reply src/components/IngredientCard/IngredientCard.jsx:13]: Ideally the them should always be provided. Even if its default. So if we leave this check here, we should still let a dev know that we did not receive a theme here and therefore used default

File: src/assets/json/themes.json (hunk 1 above)
Comment (vreddi) [reply src/assets/json/themes.json]: resolved.

File: src/components/Dashboard/Dashboard.jsx (hunk 3 above)
Comment (vreddi) [reply src/components/Dashboard/Dashboard.jsx:16]: resolved.

File: src/components/Dashboard/Dashboard.jsx (hunk 4 above)
Comment (vreddi) [reply src/components/Dashboard/Dashboard.jsx:21]: resolved.

File: src/components/IngredientCard/IngredientCard.jsx (hunk 5 above)
Comment (vreddi) [reply src/components/IngredientCard/IngredientCard.jsx:13]: Sure. But the main purpose this solves is that the `theme` comes from the backend. For some reason if the backend does not resolve a theme (data loss, some bug etc) the default theme would be used in that case also.

Comment (vreddi) [reply src/components/IngredientCard/IngredientCard.jsx:13]: Marking this as resolved as I feel a sufficient amount of discussion has been done on this.

File: src/components/IngredientCard/IngredientCard.jsx (hunk 10 above)
Comment (vreddi) [reply src/components/IngredientCard/IngredientCard.jsx]: New line is more cleaner.

File: src/components/IngredientCard/IngredientCard.scss (hunk 7 above)
Comment (vreddi) [reply src/components/IngredientCard/IngredientCard.scss]: Talked offline. Not needed.

File: src/components/IngredientCard/IngredientCard.scss (hunk 9 above)
Comment (vreddi) [reply src/components/IngredientCard/IngredientCard.scss]: Talked offline. Comment in the wrong place.

File: src/components/Chip/Chip.jsx (hunk 17 above)
Comment (vreddi) [reply src/components/Chip/Chip.jsx]: done

**Review by sukritchhabra — APPROVED** [review]

**Pull request merged**